### ISSUE TITLE
Bug: Possible regression in v3.5.5 - Rules must be an object with at least an id property

### ISSUE THREAD

**Issue by darrinmn9** [dequelabs/axe-core#2319]
Expectation: axe configurationOptions working in 3.5.4 should work in 3.5.5

Actual: In v3.5.5, I get an uncaught TypeError on page load

<pre><code>
import axe from 'axe-core'

const configurationOptions = {
  rules: axe.getRules(['wcag2aa', 'wcag2a', 'best-practice'])
}

// works on v3.5.4 
// TypeError in v3.5.5

</code></pre>

Possible regression could be from this merged commit in the release log: https://github.com/dequelabs/axe-core/commit/8c91ead900fd3cab58f77af45cce54d02e734a8e

console Error:

Uncaught TypeError: Configured rule {"ruleId":"accesskeys","description":"Ensures every accesskey attribute value is unique","help":"accesskey attribute value must be unique","helpUrl":"https://dequeuniversity.com/rules/axe/3.5/accesskeys?application=axeAPI","tags":["best-practice","cat.keyboard"]} is invalid. Rules must be an object with at least an id property

**Comment by straker** [comment]
The code by itself shouldn't throw the error. Are you passing that object directly into `axe.configure`? If so, I'm not sure that code ever worked as you intended. Without an `id` property, nothing would be configured about the rules (we never look at `ruleId`).

What exactly are you trying to do with that configuration?

**Comment by darrinmn9** [comment]
Ah, sorry for the lack of detail. Maybe it's an issue with how the configuration object is used in `vue-axe` ?https://github.com/vue-a11y/vue-axe

```
// my code

import axe from 'axe-core'
const VueAxe = require('vue-axe').default
const myConfig = {
    config: {
        rules: axe.getRules(['wcag2aa', 'wcag2a', 'best-practice'])
    }
}
  Vue.use(VueAxe, config)
```

From looking at the source it seems `vue-axe` merges the following `defaultOptions` https://github.com/vue-a11y/vue-axe/blob/master/src/utils.js#L9 with the `myConfig` object I pass.

```
// vue-axe index.js
  options = merge(defaultOptions, options)
  axeCore.configure({ ...options.config })
```

**Comment by straker** [comment]
It looks like they are passing in your object correctly to `axe.configure`, but I'm not sure your object ever did anything in the configuration.

**Comment by darrinmn9** [comment]
hmm, ok, so maybe `{rules: axe.getRules(['wcag2aa', 'wcag2a', 'best-practice'])}` has always been an incorrect config to pass to `axe-core`, but it never threw an error till v3.5.5?

**Comment by straker** [comment]
That's what I'm thinking. If you only want to run 'wcag2aa', 'wcag2a', 'best-practice' rules, then a better option would be to use the [`runOnly` option of `axe.run`](https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#options-parameter):

```js
axe.run({
  runOnly: ['wcag2aa', 'wcag2a', 'best-practice']
});
```

**Comment by darrinmn9** [comment]
Thanks @straker, my apologies for not reading the docs thoroughly.

**Comment by straker** [comment]
Closing as it seems to have been resolved. Please reopen if you fell otherwise.
